x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/ClientVersion.java

```diff
@@ -87,7 +88,11 @@ static Manifest getManifest(URL url) throws IOException {
             if (url.getProtocol().equals("jar")) {
                 JarURLConnection jarConn = (JarURLConnection) conn;
                 if (jarConn.getEntryName() == null) { // the URL points to a JAR file
-                    return jarConn.getManifest(); // in case of a fat JAR, this would return the outermost JAR's manifest
+                    Manifest manifest = jarConn.getManifest(); // in case of a fat JAR, this would return the outermost JAR's manifest
+                    try {
+                        jarConn.getJarFile().close(); // prevent locked file errors in Windows. The Manifest has been read by now
+                    } catch (IOException e) {/* ignore any closing failure */}
+                    return manifest;
```

Comment:
Shouldn't this be in a `finally` block to return not only in case of exception?

Reply:
Not sure I get your suggestion: the function returns irrespective of an `IOException` being raised. Or are you rather suggesting to return a version (in the calling function), also when an `IOExceptoin` is raised (which we currently propagate into a `IllegalArgumentException`)?

Reply:
My bad, sorry, misread that the return is within the catch block.